cmd/launcher/doctor.go

```diff
@@ -0,0 +1,541 @@
+		return fmt.Sprintf("Platform: %s", os), nil
+	}
+	return "", fmt.Errorf("Unsupported platform: %s", os)
+}
+
+type launcherFile struct {
+	name  string
+	found bool
+}
+
+// checkupRootDir tests for the presence of important files in the launcher root directory
+func checkupRootDir(filepaths []string) (string, error) {
+	importantFiles := []*launcherFile{
+		{
+			name: "debug.json",
+		},
+		{
+			name: "launcher.db",
+		},
+		{
+			name: "osquery.db",
+		},
+	}
+
+	if filepaths != nil && len(filepaths) > 0 {
+		for _, fp := range filepaths {
+			for _, f := range importantFiles {
+				if filepath.Base(fp) == f.name {
+					f.found = true
+				}
+			}
+		}
+	}
+
+	var failures int
+	for _, f := range importantFiles {
+		if f.found {
+			good(f.name)
+		} else {
+			bad(f.name)
+			failures = failures + 1
+		}
+	}
+
+	if failures == 0 {
+		return "Root directory files found", nil
+	}
+
+	return "", fmt.Errorf("%d root directory files not found", failures)
+}
+
+// checkupConnectivity tests connections to Kolide cloud services
+func checkupConnectivity(logger log.Logger, k types.Knapsack) (string, error) {
+	var failures int
+	checkpointer := checkpoint.New(logger, k)
+	connections := checkpointer.Connections()
+	for k, v := range connections {
+		if v == "successful tcp connection" {
+			good(fmt.Sprintf("%s - %s", k, v))
+		} else {
+			bad(fmt.Sprintf("%s - %s", k, v))
+			failures = failures + 1
+		}
+	}
+
+	ipLookups := checkpointer.IpLookups()
+	for k, v := range ipLookups {
+		valStrSlice, ok := v.([]string)
+		if ok && len(valStrSlice) > 0 {
+			good(fmt.Sprintf("%s - %s", k, valStrSlice))
+		} else {
+			bad(fmt.Sprintf("%s - %s", k, valStrSlice))
+			failures = failures + 1
+		}
+	}
+
+	notaryVersions, err := checkpointer.NotaryVersions()
+	if err != nil {
+		bad(fmt.Errorf("could not fetch notary versions: %w", err))
+		failures = failures + 1
+	}
+
+	for k, v := range notaryVersions {
+		// Check for failure if the notary version isn't a parsable integer
+		if _, err := strconv.ParseInt(v, 10, 32); err == nil {
+			good(fmt.Sprintf("%s - %s", k, v))
+		} else {
+			bad(fmt.Sprintf("%s - %s", k, v))
+			failures = failures + 1
+		}
+	}
+
+	if failures == 0 {
+		return "Successfully communicated with Kolide", nil
+	}
+
+	return "", fmt.Errorf("%d failures encountered while attempting communication with Kolide", failures)
+}
+
+// checkupVersion tests to see if the current launcher version is up to date
+func checkupVersion(v version.Info) (string, error) {
+	info(fmt.Sprintf("Current version: %s", v.Version))
+	// TODO: Query TUF for latest available version for this launcher instance
+	warn(fmt.Sprintf("Target version: %s", "Unknown"))
+
+	// TODO: Choose failure based on current >= target
+	if true {
+		return "Up to date!", nil
+	}
+
+	return "", fmt.Errorf("launcher is out of date")
+}
+
+// checkupConfigFile tests that the config file is valid and logs it's contents
```

Comment:
How does this validate the contents?

Reply:
Added a `PlainParser` to actually validate now, thanks!